easybuild/tools/modules.py

```diff
@@ -1469,7 +1475,17 @@ def prepend_module_path(self, path, set_mod_paths=True, priority=None):
         # Lmod pushes a path to the front on 'module use', no need for (costly) 'module unuse'
         modulepath = curr_module_paths()
         if not modulepath or os.path.realpath(modulepath[0]) != os.path.realpath(path):
+            # If no explicit priority is set, but priorities are already in use we need to use a high
+            # priority to make sure the path (very likely) ends up at the front
+            if priority is None and os.environ.get('__LMOD_Priority_MODULEPATH'):
+                priority = self.HIGH_PRIORITY
             self.use(path, priority=priority)
+            modulepath = curr_module_paths(clean=False)
+            path_idx = modulepath.index(path)
+            if path_idx != 0:
+                self.log.warn("Path '%s' could not be prepended to $MODULEPATH. The following paths have a higher "
```

Comment:
Make this `print_warning` so it pops out better? You can pass `self.log` via `log=` so the warning is logged too.

Reply:
True, done